aii-core/src/main/perl/Shellfe.pm

```diff
@@ -0,0 +1,955 @@
+         DEFAULT => undef },
+
+       { NAME    => 'boot=s',
+         HELP    => 'Node(s) to boot from local disk (can be a regexp)',
+         DEFAULT => undef },
+
+       { NAME    => 'bootlist=s',
+         HELP    => 'File with the nodes to boot from local disk',
+         DEFAULT => undef },
+
+       { NAME    => 'install=s',
+         HELP    => 'Nodes(s) to be installed (can be regexp)',
+         DEFAULT => undef },
+
+       { NAME    => 'installlist=s',
+         HELP    => 'File with the nodes to be installed',
+         DEFAULT => undef },
+
+       { NAME    => 'rescue=s',
+         HELP    => 'Node(s) to be booted in rescue mode',
+         DEFAULT => undef },
+
+       { NAME    => 'rescuelist=s',
+         HELP    => 'File with the nodes to be booted in rescue mode',
+         DEFAULT => undef },
+
+       { NAME    => 'include=s',
+         HELP    => 'Directories to add to include path',
+         DEFAULT => '' },
+
+       { NAME    => 'status=s',
+         HELP    => 'Report current boot/install status for the node ' .
+         '(can be a regexp)',
+         DEFAULT => undef },
+
+       { NAME    => 'statuslist=s',
+         HELP    => 'File with the nodes to report boot/install status',
+         DEFAULT => undef },
+
+       { NAME    => 'firmware=s',
+         HELP    => 'Nodes(s) to have their firmware image updated ' .
+                     '(can be a regexp)',
+         DEFAULT => undef },
+
+       { NAME    => 'firmwarelist=s',
+         HELP    => 'File with the nodes requiring firmware upgrade',
+         DEFAULT => undef },
+
+       { NAME    => 'livecd=s',
+         HELP    => 'Node(s) to boot to a livecd target ' .
+                    '(can be a regexp)',
+         DEFAULT => undef },
+
+       { NAME    => 'livecdlist=s',
+         HELP    => 'File with the nodes requiring livecd target',
+         DEFAULT => undef },
+
+       { NAME    => 'cdburl=s',
+         HELP    => 'URL for CDB location',
+         DEFAULT => undef },
+
+       { NAME    => 'localnclcache=s',
+         HELP    => 'Local cache for NCL library',
+         DEFAULT => '/usr/lib/aii/aii_ncl_cache' },
+
+       { NAME    => 'file=s',
+         HELP    => 'File with the nodes and the actions to perform',
+         DEFAULT => undef },
+
+         # aii-* parameters
+
+       { NAME    => 'nodiscovery|nodhcp',
+         HELP    => 'Do not update discovery (e.g. dhcp) configuration',
+         DEFAULT => undef },
+
+       { NAME    => 'nonbp',
+         HELP    => 'Do not update Network Boot Protocol (e.g. pxe) configuration',
+         DEFAULT => undef },
+
+       { NAME    => 'noosinstall',
+         HELP    => 'Do not update OS installer (e.g. kickstart) configuration',
+         DEFAULT => undef },
+
+       { NAME    => "$PROTECTED_OPTION=s",
+         HELP    => 'confirm when removing/configuring/(re)installing protected nodes. A node is protected when /system/aii/protected is set, and its value is the confirmation to use.',
```

Comment:
How about: ```required when removing/configuring/(re)installing protected nodes otherwise the operation will fail. A node is protected when /system/aii/protected is set, and the value of that path must be passed to this parameter.```

Which makes me wonder: does this only accept a single argument or can multiple values of protected be passed e.g. as a list?